### PR TITLE
Similar problem to yum#11 with $releasever being inconsistent across EL

### PR DESCRIPTION
**Pull request by agh**
Believe this moves this cookbook over to something which will consistently work.
https://github.com/chef-cookbooks/yum-chef/pull/11

### CODE REVIEW

**Comment by agh** [comment]
Nudge. This is currently still causing explosions on Oracle Linux.

```
    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    execute[yum-makecache-erlang_solutions] (/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.14.89/lib/chef/provider/yum_repository.rb l
ine 65) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
    ---- Begin output of yum -q -y makecache --disablerepo=* --enablerepo=erlang_solutions ----
    STDOUT:
    STDERR: http://packages.erlang-solutions.com/rpm/centos/7Server/x86_64/repodata/repomd.xml: [Errno 14] HTTP Error 404 - Not Found
    Trying other mirror.
```

File: attributes/erlang_solutions.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
-default['yum']['erlang_solutions']['baseurl'] = 'http://packages.erlang-solutions.com/rpm/centos/$releasever/$basearch'
-default['yum']['erlang_solutions']['description'] = 'Centos $releasever - $basearch - Erlang Solutions'
+default['yum']['erlang_solutions']['baseurl'] = "http://packages.erlang-solutions.com/rpm/centos/#{platform_version.split('.').first}/$basearch"
```
Comment (tas50) [inline attributes/erlang_solutions.rb]: Is there a reason not to just do platform_version.to_i here?

Comment (agh) [inline attributes/erlang_solutions.rb]: I copied the approach from https://github.com/chef-cookbooks/yum-chef/pull/11 which was experiencing the same failure case as this Pull Request, feeling if you merged that, it was the "approved way".

Comment (agh) [inline attributes/erlang_solutions.rb]: In essence you're asking should it be "7" or "7.2" though, both should work, however, they will exhibit different behaviour in "point release upgrade" situations -

* Using `.to_i` means you'd pull from /7.2 and point upgrades are on the user, this cookbook will only upgrade to /7.3 after the system is also upgraded to that point release.
* Using `.split` means you'd pull from /7 and Chef is likely to **immediately** pick up any packages which are newer and distributed with 7.3 when that is released.

I'm happy to do either but I think if we go with `platform_version.to_i` that you should audit other community cookbooks, including Yum, and converge the approach. I think differing behaviours in different cookbooks would potentially be a poor user experience. /cc @iennae

**Comment by agh** [comment]
I'm closing this one out as I'm unable to commit further time to this Pull Request.

**Comment by iennae** [comment]
@agh If you reopen, I think this is fine to merge in. We should make sure that everything is consistent across the different repositories since as @afiune pointed out, EL is not consistent and this is causing issues.

**Comment by agh** [comment]
I actually can't reopen because I cleaned up the source repository a long time ago.

Apparently that doesn't hurt you if you hit <MERGE> but if you <CLOSE> and then want to <REOPEN> you're basically hosed.

screen shot 2016-11-14 at 17 41 57

Not sure how to solve this right now, I suspect I'll have to recreate from scratch.

**Comment by iennae** [comment]
Ahh. I'll just recreate. Thanks!

**Pull request closed without merging**